modules/javafx.graphics/src/main/java/javafx/scene/text/Text.java

```diff
@@ -1044,6 +1036,49 @@ public final HitInfo hitTest(Point2D point) {
         return new HitInfo(h.getCharIndex(), h.getInsertionIndex(), h.isLeading());
     }
 
+    private int findRunIndex(double x, double y, GlyphList[] runs) {
+        int runIndex = 0;
```

Comment:
some general comment for this method:
1. there are many places where runs[runIndex] is repeated within the same code block, I wonder if it would make sense to create a local variable.  It might be tricky because the runIndex changes mid-flight.
2. perhaps `runIndex` could be shortened to `ix` to make the lines shorter?

Reply:
Refactored the function. Changed `runIndex` to `ix`. 
I could remove some repetition of `runs[runIndex]`. Had to keep some instances since previous or next  runs are used in some cases.